libraries-llms/src/test/java/com/baeldung/langchain/ChatWithMemoryLiveTest.java

```diff
@@ -0,0 +1,45 @@
+package com.baeldung.langchain;
+
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.memory.ChatMemory;
+import dev.langchain4j.memory.chat.TokenWindowChatMemory;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.openai.OpenAiChatModel;
+import dev.langchain4j.model.openai.OpenAiTokenizer;
+
+import static dev.langchain4j.data.message.UserMessage.userMessage;
+import static dev.langchain4j.model.openai.OpenAiModelName.GPT_3_5_TURBO;
+
+import java.util.logging.Logger;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ChatWithMemoryLiveTest {
+
+    @Test
+    public void givenMemory_whenPrompted_thenValidResponse() {
+
+        ChatLanguageModel model = OpenAiChatModel.withApiKey(Constants.OPEN_API_KEY);
+        ChatMemory chatMemory = TokenWindowChatMemory.withMaxTokens(300, new OpenAiTokenizer(GPT_3_5_TURBO));
+
+        chatMemory.add(userMessage("Hello, my name is Kumar"));
+        AiMessage answer = model.generate(chatMemory.messages())
+            .content();
+        Logger.getGlobal()
+            .info(answer.text());
+        Assert.assertNotNull(answer.text());
+        chatMemory.add(answer);
+
+        chatMemory.add(userMessage("What is my name?"));
+        AiMessage answerWithName = model.generate(chatMemory.messages())
+            .content();
+        Logger.getGlobal()
+            .info(answerWithName.text());
+        Assert.assertTrue(answerWithName.text()
+            .contains("Kumar"));
+        chatMemory.add(answerWithName);
```

Comment:
Highlighting this again

Reply:
I am not sure why this conversation was hidden previously, sorry to miss it! The reason I have kept it is to show that we have to keep adding the responses back to the memory to keep building the context. The conversation itself can be much longer than what has been shown here. I have moved the assertion to the end so that it does not look redundant.